services/admin/src/store.ts

```diff
@@ -96,4 +135,677 @@ export class Store {
       id
     );
   }
+
+  /**
+   * Asserts that an election with the given ID exists and is not deleted.
+   */
+  assertElectionExists(electionId: Id): void {
+    const election = this.client.one(
+      `
+        select id from elections
+        where id = ? and deleted_at is null
+      `,
+      electionId
+    ) as { id: Id } | undefined;
+
+    if (!election) {
+      throw new Error(`Election not found: ${electionId}`);
+    }
+  }
+
+  /**
+   * Adds a CVR file record and returns its ID. If a CVR file with the same
+   * contents has already been added, returns the ID of that record instead.
+   *
+   * Call with `analyzeOnly` set to `true` to only analyze the file and not add
+   * it to the database.
+   */
+  addCastVoteRecordFile({
+    electionId,
+    filename,
+    cvrFile,
+    analyzeOnly,
+  }: {
+    electionId: Id;
+    filename: string;
+    cvrFile: string;
+    analyzeOnly?: boolean;
+  }): Result<
+    {
+      id: Id;
+      wasExistingFile: boolean;
+      newlyAdded: number;
+      alreadyPresent: number;
+    },
+    AddCastVoteRecordError
+  > {
+    this.assertElectionExists(electionId);
```

Comment:
this is not needed as the foreign key constraint will fail the row insert if the referenced `election_id` doesn't exist.